File: pipeline/detectors/diff_no_such_file.py
"""
Detector for diff errors when a file is missing.
"""

import re
import typing as T
from pipeline.detectors.base import Detector
from pipeline.models import ErrorClue


class DiffNoSuchFileDetector(Detector):
    """
    Detect diff errors when a file is missing.

    Matches patterns like:
    - diff: test.txt: No such file or directory
    """

    PATTERNS = {
        "missing_file": r"diff:\s*(?P<file_path>[^\s:]+):\s*No such file or directory",
    }

    EXAMPLES = [
        (
            "diff: test.txt: No such file or directory",
            {
                "clue_type": "missing_file",
                "confidence": 1.0,
                "context": {"file_path": "test.txt"},
            },
        ),
    ]

File: src/boil/detectors/diff_no_such_file.go
// Detector for diff command "No such file" errors.
package detectors

// DiffNoSuchFileDetector detects diff command "No such file" errors
type DiffNoSuchFileDetector struct {
	*BaseDetector
}

func NewDiffNoSuchFileDetector() (*DiffNoSuchFileDetector, error) {
	base, err := NewBaseDetector(
		"DiffNoSuchFileDetector",
		100,
		map[string]string{
			"missing_file": `diff:\s*(?P<file_path>[^\s:]+):\s*No such file or directory`,
		},
		[]DetectorExample{
			{
				Name:     "diff_missing_file",
				Input:    "diff: expected.txt: No such file or directory",
				ClueType: "missing_file",
				Context:  map[string]string{"file_path": "expected.txt"},
			},
		},
	)
	if err != nil {
		return nil, err
	}
	return &DiffNoSuchFileDetector{base}, nil
}
